api/src/test/java/org/openmrs/module/dhisreport/api/db/hibernate/HibernateDHIS2ReportingDAOTest.java

```
@@ -0,0 +1,290 @@
+    extends BaseModuleContextSensitiveTest
+{
+    protected final Log log = LogFactory.getLog( getClass() );
+
+    protected static final String INITIAL_OBS_XML = "db/ObsServiceTest-initial.xml";
+
+    DHIS2ReportingService dhis2ReportingService = null;
+
+    DHIS2ReportingDAO dao;
+
+    @Before
+    public void before()
+        throws Exception
+    {
+        executeDataSet( INITIAL_OBS_XML );
+        dhis2ReportingService = Context.getService( DHIS2ReportingService.class );
+        dao = (DHIS2ReportingDAO) dhis2ReportingService.getDao();
+
+    }
+
+    @Test
+    public void shouldevaluateDataValueTemplate()
+        throws DHIS2ReportingException, JAXBException, IOException, ParseException
+    {
+        ClassPathResource resource = new ClassPathResource( "reportDefinition.xml" );
+        JAXBContext jaxbContext = JAXBContext.newInstance( ReportTemplates.class );
+
+        Unmarshaller jaxbUnmarshaller = jaxbContext.createUnmarshaller();
+        ReportTemplates reportTemplates = (ReportTemplates) jaxbUnmarshaller.unmarshal( resource.getInputStream() );
+        List<ReportDefinition> rds = reportTemplates.getReportDefinitions();
+        String timeperiod = "2016-06-28";
+        Location location = new Location();
+        location.setName( "County General" );
+        location.setDescription( "desc" );
+        location.setAddress1( "address1" );
+        location.setId( 123 );
+
+        Period period = new MonthlyPeriod( new SimpleDateFormat( "yyyy-MM-dd" ).parse( timeperiod ) );
+        SimpleDateFormat dateformat = new SimpleDateFormat( "yyyy-MM-dd" );
+        String start = "2016-06-28";
+        String end = "2016-06-30";
+        Date startDate = null;
+        Date endDate = null;
+        try
+        {
+            startDate = dateformat.parse( start );
+            endDate = dateformat.parse( end );
+        }
+        catch ( ParseException e )
+        {
+            e.printStackTrace();
```

Instead of catching this exception, shouldn't you let it bubble through?